### DISCUSSION TITLE
FileUploadHandler didn't handle long original filenames.

### DISCUSSION THREAD

**Discussion by ZerglingGo** [livewire/livewire#3021] [7. Bugs (archived)]
### Description
- Related discussion #2238 #2813 
> When uploads with a long filename, it throws an exception:
> `fopen(<path with long filename>): failed to open stream: File name too long`

- In the process of encoding the original filename to base64.
- Increases filename length compared to the original filename.
https://github.com/livewire/livewire/blob/b0f4026509604f5239813a605ba7172f4bb5fa0e/src/TemporaryUploadedFile.php#L126-L133
- FileUploadHandler isn't catching that increases filename could be over the maximum length of filename.
https://github.com/livewire/livewire/blob/b0f4026509604f5239813a605ba7172f4bb5fa0e/src/Controllers/FileUploadHandler.php#L32-L42
### Exact steps to reproduce
- Select a file with a long filename in the file input form.

### Stripped-down, copy-pastable code snippets

UploadTest.php
```php
<?php

namespace App\Http\Livewire;

use Livewire\Component;
use Livewire\WithFileUploads;

class UploadTest extends Component
{
    use WithFileUploads;

    protected $rules = [
        'file' => 'required',
    ];

    public $file;

    public function render()
    {
        return view('livewire.upload-test');
    }
}
```

upload-test.blade.php
```html
<div>
    <input wire:model="file" type="file">
</div>
```

**Context**
- Livewire version: 2.4.3
- Laravel version: 8.39.0
- Alpine version: 2.8.2
- Browser: Chrome

**Comment by fodor11** [comment]
Hey @calebporzio! We would really appreciate if this problem was addressed. Is there any chance of a fix incoming in the near future?

**Comment by faydanube** [comment]
Hello, @ZerglingGo Is this problem solved?

**Reply by faydanube** [reply]
There are two ways of thinking:
1, verify the length of the file name after the user selects the file and throw an error to alert the user.
2, Customise the logic of livewire upload.

**Reply by ZerglingGo** [reply]
Not merged, but there are a few fix PRs.
- https://github.com/livewire/livewire/pull/2841
- https://github.com/livewire/livewire/pull/8690

**Reply by Naoray** [reply]
not yet. Got a PR rdy for merge to solve this https://github.com/livewire/livewire/pull/9172
